### ISSUE TITLE
Public Staticman server is hitting GitHub API limits

### ISSUE THREAD

**Issue by maxime-michel** [mmistakes/minimal-mistakes#1842]
I would like to raise awareness about recent issues me and others have experienced with Staticman.

The first is that it is becoming almost impossible to get an 'OK' back when adding the bot as a repository collaborator: https://github.com/eduardoboucas/staticman/issues/227.

The second is that some comments might possibly get lost due to the public instance (`api.staticman.net`) hitting GitHub API limits: https://github.com/eduardoboucas/staticman/issues/222

The solution to those problems is to allow people to define a custom URL for their own deployment of Staticman. But I also believe some awareness about the potential issues should be present in the documentation.

Another idea could be that the theme transparently uses its own deployment of Staticman. It would greatly reduce the simplicity if we had one instance to host N users, rather than N users paying for and setting up one instance individually. But somebody's got to handle it at the end of the day, and that would have to be well organized.

Any thoughts?

PS: this is an excellent theme and a great framework! I wish I had joined the party sooner.

**Comment by mmistakes** [comment]
Thanks for bringing it up. Not sure on the best path forward.
I guess anyone who relies on the public instance probably needs a backup plan. Kind of comes with the territory when using "free" stuff.

**Comment by maxime-michel** [comment]
Would you be OK with a PR that opens the endpoint URL to possible configuration?

**Comment by mmistakes** [comment]
Yeah I'd be cool with that. Could probably do something like `{{ site.staticman.endpoint | default: 'https://api.staticman.net/v2/entry/' }}` to fallback to the public instance if one isn't specified in `_config.yml`.
